### ISSUE TITLE
Excel_Toolkit: Concatenate lists on input of multiple cells with lists in

### ISSUE THREAD

**Issue by rwemay** [BHoM/Excel_UI#98]
When referencing multiple cells with lists (of objects) in, the lists should be concatenated. At the moment this just returns an error.

**Comment by awakeman** [comment]
I am not certain it is necessarily a good idea to do this automatically, there are doubtless edge cases where it would not be what the user wants. Obvious edge cases being when the parameter type is `object` or `List<object>` and the user *wants* to maintain a list of lists as the input to those parameters.

I propose an Engine method to concatenate lists to give the user control over this, alternatively be even smarter and attempt to abstract and expose Linq.

Thoughts.

**Comment by rwemay** [comment]
Agree! And as discussed briefly earlier, I thought I'd cancelled this before hitting the green button as I realised as I was typing it that we wouldn't want a general method to squash lists of anything together. A specific concatenate method would be great. Also really good if we can bottom out the methodology/philosophy for reflecting Linq into the BHoM, as this will be massively useful for things like Excel which do not have good list/collection-management functions (without rendering masses of data). 

@al-fisher @adecler , @IsakNaslundBh , good to get this Linq plan/discussion going in the Engine. 

Closing this issue here for now and I suggest no action until we bottom out Linq plan.
